### PR TITLE
Initial Commit from OCL

### PR DESCRIPTION
**Pull request by deeopencitieslab**
… image recg.

### CODE REVIEW

File: service-request-infno-spec-1.0.0-swagger.json [hunk 1]
```diff
@@ -0,0 +1,213 @@
+{
+  "openapi" : "3.0.0",
+  "servers" : [ {
+    "url" : "https://qaeservices1.capetown.gov.za/coct/api"
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: remove for now - we'll add these when I've reserved a domain for this

File: service-request-infno-spec-1.0.0-swagger.json [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  } ],
+  "info" : {
+    "description" : "This is a example API for Service Request.",
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: `"This is a radically open, non-authed API intended to provide information to the residents of Cape Town"`

File: service-request-infno-spec-1.0.0-swagger.json [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      "get" : {
+        "tags" : [ "3rd_parties" ],
+        "summary" : "Get service request API Info",
+        "parameters" : [ {
+          "in" : "query",
+          "name" : "service_request_no",
+          "description" : "maximum number of records to return",
+          "required" : true,
+          "schema" : {
+            "type" : "integer",
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: rather make this a string, please.

Please also add validation of it being a 10 digit number, starting with `9`

File: service-request-infno-spec-1.0.0-swagger.json [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+          "502" : {
+            "description" : "Bad Gateway Error"
+          },
+          "503" : {
+            "description" : "Service Unavailable Error"
+          }
+        }
+      }
+    },
+    "/service-request/image_placeholder/" : {
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: rather make this path, `image_classification`

File: service-request-infno-spec-1.0.0-swagger.json [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+      "get" : {
+        "tags" : [ "media" ],
+        "parameters" : [ {
+          "in" : "query",
+          "name" : "labels",
+          "required" : false,
+          "schema" : {
+            "type" : "string"
+          }
+        } ],
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: for the parameters, we would want to be passing an image in the body of the request - [this is the relevant bit of the docs](https://swagger.io/docs/specification/describing-request-body/file-upload/)

File: service-request-infno-spec-1.0.0-swagger.json [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+          }
+        }
+      },
+      "ImageRecognitionSchema" : {
+        "required" : [ "url" ],
+        "properties" : {
+          "url" : {
+            "type" : "array"
+          }
+        }
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: I think our response here will be an array of objects of the form:
```json
[
  {
  "type": "string",
  "confidence": "float",
  }
]
```

File: service-request-infno-spec-1.0.0-swagger.json [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+            "maximum" : 9
+          }
+        } ],
+        "responses" : {
+          "200" : {
+            "description" : "OK",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                  "type" : "array",
```
Comment (Gordonei) [inline service-request-infno-spec-1.0.0-swagger.json]: I think we would only ever return one `ServiceRequestSchema` from this path

**Review by Gordonei — CHANGES_REQUESTED** [review]
Looks good - very much in line with what we discussed in the design workshop. I might take another pass tomorrow, after looking at some of my notes.

File: service-request-infno-spec-1.0.0-swagger.json (hunk 1 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: Removed

File: service-request-infno-spec-1.0.0-swagger.json (hunk 2 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: Description has been updated

File: service-request-infno-spec-1.0.0-swagger.json (hunk 3 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: input type changed

File: service-request-infno-spec-1.0.0-swagger.json (hunk 4 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: path changed

File: service-request-infno-spec-1.0.0-swagger.json (hunk 6 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: Response has been changed

File: service-request-infno-spec-1.0.0-swagger.json (hunk 5 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: image_classification field type changed

File: service-request-infno-spec-1.0.0-swagger.json (hunk 7 above)
Comment (deeopencitieslab) [reply service-request-infno-spec-1.0.0-swagger.json]: Please elaborate, I don't think I fully understand

Comment (Gordonei) [reply service-request-infno-spec-1.0.0-swagger.json]: so the return type of this path can just be:

```json
"content" : {
  "application/json" : {
    "schema" : {
      "$ref":  "#/components/schemas/ServiceRequestSchema"
    }
  }
}
```

**Pull request merged**